var/spack/repos/builtin/packages/py-pyem/package.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyPyem(PythonPackage):
+    """Project-level Python virtual environment management tool.
+
+    PyEM manages multiple virtual environments local to projects.
+    It provides shortcuts to create, remove, switch between, and run
+    commands against virtual environments created against
+    various Python interpreters."""
+
+    homepage = "https://pypi.org/project/pyem/#description"
+    pypi     = "pyem/pyem-2.1.0.tar.gz"
+
+    version('2.1.0', sha256='5234a20427ab2813a8a0bf1e9112d4d854b1b0502b3e63d17c1b1a3c4be9340e')
+
+    depends_on('python@3.7:', type=('build', 'run'))
+    depends_on('py-virtualenv', type='build')
+    depends_on('py-pip', type='build')
+    depends_on('py-pytest', type=('build', 'run'))
+    depends_on('py-mypy', type=('build', 'run'))
+    depends_on('py-black', type=('build', 'run'))
+    depends_on('flit', type=('build', 'run'))
+
+    def install(self, spec, prefix):
+        pip = which('pip')
+        pip('install', self.stage.archive_file, '--prefix={0}'.format(prefix))
```

Comment:
The default install method should work fine, you shouldn't need to override this.

Reply:
Without overriding the install process none of the pyem commands are recognized. This seemed to be fixed when I added the dep for py-pip and overrode the install phase.

Reply:
> none of the pyem commands are recognized

Can you elaborate on this? How can I reproduce this issue?

Reply:
`spack install py-pyem`
`module load py-pyem`
`module list` *verify the proper module is loaded
`pyem venv add python3.7`
returns "-bash: pyem: command not found"

Reply:
Ah, it looks like their build script is broken. The following patch solved the issue for me:
```diff
--- a/setup.py	2021-03-16 16:30:13.000000000 -0500
+++ b/setup.py	2021-03-16 16:30:41.000000000 -0500
@@ -1,7 +1,7 @@
 #!/usr/bin/env python
 # setup.py generated by flit for tools that don't yet use PEP 517
 
-from distutils.core import setup
+from setuptools import setup
 
 packages = \
 ['pyem', 'pyem.projects']
```
You'll also need to add a build/run dep on setuptools. If you have the time, please try to contribute that patch upstream. If not, I can try.

Reply:
Under what file did you find `a/setup.py	2021-03-16 16:30:13.000000000 -0500` and `from distutils.core import setup`? This is a minor package for me so I'll probably just filter_file it out unless you want to fix it upstream.

Reply:
This patch applies to the `setup.py` file.

Reply:
I just used filter_file and it seems to install and work fine now.